nautobot/core/api/views.py

```diff
@@ -702,3 +709,90 @@ def execute_graphql_request(self, request, data, query, variables, operation_nam
             return document.execute(**options)
         except Exception as e:
             return ExecutionResult(errors=[e], invalid=True)
+
+
+#
+# Lookup Expr
+#
+
+
+class GetFilterSetFieldLookupExpressionChoicesAPI(NautobotAPIVersionMixin, APIView):
+    """API View that gets all lookup expression choices for a FilterSet field."""
+
+    permission_classes = [IsAuthenticated]
+
+    @extend_schema(exclude=True)
+    def get(self, request):
+        if "content_type" not in request.GET or "field_name" not in request.GET:
+            return Response(
+                "content_type and field_name are required parameters",
+                status=400,
+            )
+        contenttype = request.GET.get("content_type")
+        field_name = request.GET.get("field_name")
```

Comment:
Just thinking aloud here, what do you think about maybe making these part of the URL path instead of query-parameters, since they're required? I.e. `/api/core/filterset-fields/<app_label>.<model>/<field_name>/choices/` and `/api/core/filterset-fields/<app_label>.<model>/<field_name>/dom-element/`?

Reply:
`field_name` is obtained in the UI when the field has been selected.

Reply:
Sorry, I'm not sure I understand how that's a response to my idea?

Reply:
Here #[2348L306](https://github.com/nautobot/nautobot/pull/2348/files#diff-98f1dfde62e7c5d2714e5fc4be0a06544d109330ccfb842b063be81ec3894559R306)
```python
self.fields["lookup_type"].widget.attrs["data-url"] = reverse("core-api:filtersetfield-list-lookupchoices")
```
The endpoint is added to the `lookup_type` field, however, the `field_name` is currently unknown. The `field name` is obtained only after the user has picked the field in the UI.

Reply:
hence, using this would be virtually impossible. `reverse("...", kwargs={"model": .., "app_label":.., "field_name":...})`